### PR TITLE
Implement searchGroupingMembers endpoint

### PR DESCRIPTION
**Pull request by gitCarrot**
# Ticket Link

[Ticket-1842](https://uhawaii.atlassian.net/browse/GROUPINGS-1842)

# List of squashed commits

- Implement new grouper api service function to get a list of members with search string

- Implement the functions of service layer and controller for the new end point

- Implement the tests for new end-point and the function in service and controller, modify the test data to pass the new getSubjects function in subjectsResults class
- Rename API URL to more RESTFUL

# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Integration Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

### CODE REVIEW

**Review by mhodgesatuh — APPROVED** [review]

File: src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java [hunk 1]
```diff
@@ -541,6 +541,19 @@ public ResponseEntity<GroupingDescription> groupingDescription(
                 .body(groupingOwnerService.groupingsDescription(currentUser, path));
     }
 
+    /**
+     * Get a list of members for a selected grouping path with search string.
+     */
+    @GetMapping(value = "/groupings/{path:[\\w-:.]+}/{searchString}")
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java:547]: I know I put this on the ticket for the URL, but now I'm looking at it, it is more RESTful if the URL was:

/groupings/{path}/search/{search}

**Comment by mhodgesatuh** [comment]
agreed

Michael Hodges (he/him/his)
Enterprise Middleware, Identity and Access Management
University of Hawaii, Information Technology Services
***@***.***
808-956-7195 (GMT -10:00)

we>$

File: src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java (hunk 1 above)
Comment (gitCarrot) [reply src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java]: Sounds good for the new search grouper functions in the future

**Review by JorWo — APPROVED** [review]
Great work!

**Pull request merged**